### PR TITLE
[TEAM2-292] Fix Sentry Bugs

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
Just guarding against empty values here.

## What to test
I don't think there's a good way to test these corner cases. Let's just make sure the code is solid.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

**Review by skylarbarrera — APPROVED** [review]
gm

File: src/utils/ethereumUtils.ts [hunk 1]
```diff
@@ -117,7 +117,8 @@ const getNativeAssetForNetwork = async (
 ): Promise<ParsedAddressAsset | undefined> => {
   const networkNativeAsset = getNetworkNativeAsset(network);
   const { accountAddress } = store.getState().settings;
-  let differentWallet = address.toLowerCase() !== accountAddress.toLowerCase();
+  let differentWallet =
+    address?.toLowerCase() !== accountAddress?.toLowerCase();
```
Comment (estebanmino) [inline src/utils/ethereumUtils.ts:121]: fixed https://sentry.io/organizations/rainbow-me/issues/3349023668/?project=1855565&query=release%3Ame.rainbow%401.6.17%2B31&statsPeriod=14d

File: src/redux/data.ts [hunk 2]
```diff
@@ -1191,8 +1194,8 @@ export const addressAssetsReceived = (
   const assetsWithScamURL: string[] = Object.values(parsedAssets)
     .filter(
       asset =>
-        (isValidDomain(asset.name.replaceAll(' ', '')) ||
-          isValidDomain(asset.symbol)) &&
+        ((asset?.name && isValidDomain(asset?.name.replaceAll(' ', ''))) ||
+          (asset?.symbol && isValidDomain(asset.symbol))) &&
```
Comment (estebanmino) [inline src/redux/data.ts:1198]: fixed https://sentry.io/organizations/rainbow-me/issues/3445380812/?project=1855565&query=release%3Ame.rainbow%401.6.17%2B31&statsPeriod=14d

Comment (estebanmino) [reply src/redux/data.ts:1198]: and https://sentry.io/organizations/rainbow-me/issues/3442735980/?project=1855565&query=release%3Ame.rainbow%401.6.17%2B31&statsPeriod=14d

File: src/redux/data.ts [hunk 3]
```diff
@@ -895,10 +895,13 @@ const checkForUpdatedNonce = (transactionData: ZerionTransaction[]) => (
...
+      .filter(tx => {
+        const addressFrom = tx?.address_from;
+        return (
+          addressFrom &&
+          addressFrom.toLowerCase() === accountAddress.toLowerCase()
+        );
+      })
       .sort(({ nonce: n1 }, { nonce: n2 }) => (n2 ?? 0) - (n1 ?? 0));
     const [latestTx] = txSortedByDescendingNonce;
     const { address_from, nonce } = latestTx;
```
Comment (estebanmino) [inline src/redux/data.ts:907]: i believe this. one https://sentry.io/organizations/rainbow-me/issues/3332013511/?project=1855565&query=release%3Ame.rainbow%401.6.17%2B31&statsPeriod=14d is coming from here when there's no latestTx object

**Review by estebanmino — APPROVED** [review]

File: src/redux/data.ts (hunk 3 above)
Comment (estebanmino) [reply src/redux/data.ts]: NIIIICE
